app/scenes/select_server/select_server.js

```diff
@@ -21,22 +21,45 @@ import logo from 'assets/images/logo.png';
 import Client from 'service/client';
 import RequestStatus from 'service/constants/request_status';
 
-export default class SelectServer extends Component {
+export default class SelectServer extends PureComponent {
     static propTypes = {
-        serverUrl: React.PropTypes.string.isRequired,
-        server: React.PropTypes.object.isRequired,
-        actions: React.PropTypes.object.isRequired
+        serverUrl: PropTypes.string.isRequired,
+        server: PropTypes.object.isRequired,
+        actions: PropTypes.object.isRequired
     };
 
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            error: null
+        };
+    }
+
     onClick = () => {
-        Client.setUrl(this.props.serverUrl);
+        const url = this.props.serverUrl;
+        const regex = /^https?:\/\//i;
+        let error = null;
+
+        if (regex.test(url)) {
+            Client.setUrl(url.replace(/\/+$/, ''));
```

Comment:
No matter how simple, regexes always add a layer of opacity. Instead of taking a second or two to read this regex, I would've realized its purpose instantly if we extracted it out to a function like: `stripTrailingSlashes`. Besides, we'll likely want to do even more normalization (like trimming trailing whitespace or downcasing the protocol), which would call for being its own function.